src/main/java/avd/inf/jdm/rentmycar/controller/RideController.java

```diff
@@ -0,0 +1,67 @@
+package avd.inf.jdm.rentmycar.controller;
+
+import avd.inf.jdm.rentmycar.domain.Offer;
+import avd.inf.jdm.rentmycar.domain.Ride;
+import avd.inf.jdm.rentmycar.service.RideService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api")
+public class RideController {
+
+private final RideService rideService;
+
+@Autowired
+    public RideController(RideService rideService) { this.rideService = rideService;}
+
+    @GetMapping("/v1/rides")
+    public ResponseEntity<List<Ride>> getAllRides() {
+    List<Ride> found = rideService.getAllRides();
+    return found.isEmpty() ?  ResponseEntity.noContent().build() : ResponseEntity.ok(found);
+    }
+    @GetMapping("/v1/rides/car/{carId}")
+    public ResponseEntity<List<Ride>> getRidesById(@PathVariable Long carId) {
+        try {
+            List<Ride> found = new ArrayList<>();
+            found.addAll(rideService.getRidesByCarId(carId));
+            return found.isEmpty() ? ResponseEntity.noContent().build() : ResponseEntity.ok(found);
```

Comment:
You could implement the new ResponseHandler class in the class, but I can also do that for you if you want.

Reply:
Could we do this later? I need to work on my other cards as well :)

Reply:
Absolutely!
Also, it is not a must, just a suggestion 😄